### PR TITLE
Itm 324 aggregate graphs

### PR DESCRIPTION
**Pull request by kaitlyn-sharo**
Added new graphs for survey data visualization. I know Brian is working with this branch on his own branch, but wanted everyone to have a chance to start looking at the code so I could start making changes if needed.

Also includes package upgrades and a much needed node upgrade

To test, rebuild everything, check out all pages with specific focus on the program questions page

### CODE REVIEW

**Comment by brianpippin** [comment]
Added some charts, reworked navigation and home page.  Ready for @kaitlyn-sharo to review and @dereknop and anyone else.

**Review by brianpippin — APPROVED** [review]

**Comment by kaitlyn-sharo** [comment]
@brianpippin I love the way you styled the program questions page, but I think the last two graphs (Adept and Soartech) are supposed to be in a separate section, not grouped with question 3. There was originally a "question 4" label, but Jennifer asked me to take it away just because there was no actual question with it, but it did separate these two graphs from question 3

**Comment by dereknop** [comment]
> image Also, just a very minor scrolling problem on the human sim probe page. You can see that column 0 stays sticky, but the other columns can be seen scrolling past it. Probably some sort of padding/margin problem


https://github.com/NextCenturyCorporation/itm-evaluation-dashboard/assets/73036954/03013498-cd46-4d66-971c-65901cc919d7

This is the behavior I get when I scroll. I don't see anything scrolling past, maybe it is different browser behavior? I am on chrome.

File: dashboard-ui/src/components/AggregateResults/aggregateResults.jsx [hunk 1]
```diff
@@ -117,55 +241,106 @@ export default function AggregateResults() {
...
+                    
+                
+                    {aggregateData["groupedSim"]!== undefined && Object.keys(aggregateData["groupedSim"]).map((objectKey, key) => 
+                        <div className='chart-home-container' key={"container_" + key}>
+                            <div className='chart-header'>
+                                <div className='chart-header-label'>
+                                    <h4 key={"header_" + objectKey}>{objectKey[0].toUpperCase() + objectKey.slice(1)}</h4>
+                                </div>
+                            </div>
+                            <div key={"container_" + key} className='resultTableSection result-table-section-override'>
```
Comment (dereknop) [inline dashboard-ui/src/components/AggregateResults/aggregateResults.jsx:321]: On this line, removing 'result-table-section-override' fixed the problem @kaitlyn-sharo  mentioned for me. I think the important flags in that class in aggregateResults.css may have been causing the problem.  The only other effect I noticed from removing those styles was the rounded border on the table.

Before removing:
Screenshot 2024-04-09 at 4 33 43 PM

After Removing:
Screenshot 2024-04-09 at 4 34 44 PM

**Review by dereknop — APPROVED** [review]
The new home page looks great, cool library for those graphs. I didn't come across any other visual bugs on the other pages besides the one Kaitlyn mentioned.

File: dashboard-ui/src/components/AggregateResults/aggregateResults.jsx (hunk 1 above)
Comment (kaitlyn-sharo) [reply dashboard-ui/src/components/AggregateResults/aggregateResults.jsx:321]: Yeah, probably the padding in result-table-section-override. Maybe see if you can get the same effect with margin instead.

**Comment by brianpippin** [comment]
> @brianpippin I love the way you styled the program questions page, but I think the last two graphs (Adept and Soartech) are supposed to be in a separate section, not grouped with question 3. There was originally a "question 4" label, but Jennifer asked me to take it away just because there was no actual question with it, but it did separate these two graphs from question 3

As you probably saw with my conversation from Jennifer she wants to leave this in the same area, so no action needed.

File: dashboard-ui/src/components/AggregateResults/aggregateResults.jsx (hunk 1 above)
Comment (brianpippin) [reply dashboard-ui/src/components/AggregateResults/aggregateResults.jsx:321]: This should all be resolved now.  Please let me know.

Comment (kaitlyn-sharo) [reply dashboard-ui/src/components/AggregateResults/aggregateResults.jsx:321]: Resolved! Looks great

**Comment by kaitlyn-sharo** [comment]
I approve (can't actually approve since it's my pr)

**Pull request merged**